SIM_MODE and session ID fixes for arale

alfonsosanchezbeato opened the pull request:
A couple of fixes needed for arale

tonyespy commented:
Overall looks good.  A couple of minor comments:

1. Suggest re-wording of first commit message to "Don't send..."

2. For the second commit message:

 * should we use 'arale' or 'tangxi' in the commit message?

 * it would be helpful to add the reasoning of we need to send the session ID, and what it respresents ( ie. what we discussed on IRC ) to the detailed commit message, and probably in the code as well ( maybe just for the first instance ).

3. If feasible, we should add one or more unit tests for the session ID change.  To do so would require creating a mock GRil struct with the proper vendor set.

alfonsosanchezbeato commented:
Comments addressed and PR refreshed.

The case of unit-testing a request with MTK specific quirks was considered some time ago, and test-mtkrequest.c contains some examples. I have added one test and fixed the already existing tests, that in fact should have failed after doing this modification.

Re: arale vs tangxi, I think arale is more appropriate as it names the device, while tangxi is the project.

tonyespy commented:
Approved.
